src/query/service/tests/it/sql/planner/format/mod.rs

```diff
@@ -154,6 +154,7 @@ fn test_format() {
                 from_correlated_subquery: false,
                 need_hold_hash_table: false,
                 broadcast: false,
+                is_lateral: false,
```

Comment:
The lateral join should have been decorrelated before other optimizations.

What will we use the flag for?

Reply:
Lateral join is different from subquery, it will generate the same column index, when the filter is pushed down to the join, it cannot be judged whether it belongs to the probe or build side, so we use this flag to mark whether a join is a lateral join, we will not pull up the filter below the lateral join because we cannot push it down.

Reply:
> it will generate the same column index

It sounds like a bug. 

Would you file an issue for this? I'll take a look at it later.

Reply:
> > it will generate the same column index
> 
> It sounds like a bug.
> 
> Would you file an issue for this? I'll take a look at it later.

Yeah.